doc/development-process.md

```diff
@@ -0,0 +1,132 @@
+# Development Process
+
+Depending on your area of focus we're providing guidance of each kind of development setup you might want.
+
+Step one, you **must** have the following things installed:
+
+* [go](https://go.dev) v1.17 -- Primary development language
+* [docker](https://www.docker.com/) -- Used for generating containers & testing kubernetes set-ups.
+* [kind](https://kind.sigs.k8s.io/) -- Run kubernetes clusters in docker for testing.
+* [flux](https://fluxcd.io/) -- Continuous delivery system for kubernetes that weave-gitops enriches.
+* [tilt](https://tilt.dev/) -- Automatically build and deploy to a local cluster.
+
+Now you have those things installed, lets go!
+
+## Quick start development setup
+
+Help me get going as fast as possible!
+
+Create a kind cluster. We'll do this so we can have a local cluster on our machine
+
+`$ kind create cluster --name=<some name>` 
+
+Then we install flux on it.
+
+`$ flux install`
+
+Then we bring up tilt.
+
+`$ tilt up`
+
+Tilt will tell you the url, or hit the space bar and it'll automagically open a browser for you which will show your local cluster and the resources running. You should see 3 things, Tiltfile, dev-weave-gitops, uncategorised.
+
+Goto login with username: dev, password:dev at http://localhost:9001/sign_in
+
+Woop! It's working. It'll be empty cos you haven't created any flux objects. Follow the [Weave Gitops getting started guide](https://docs.gitops.weave.works/docs/getting-started/), which you'll create repos and flux objects.
```

Comment:
this feels weird, from the above - Weave GitOps is already installed and you can login to the web UI. This guide then asks you to install flux and weave gitops again... even if we navigate to a specific section like https://docs.gitops.weave.works/docs/getting-started/#part-2---deploying-and-viewing-podinfo-application its based on `flux bootstrap` rather than `flux install`.

Reply:
I was trying to not re-write instructions of how to have objects appear in your UI. Any suggestions? or it needs it's own, just do this.

Reply:
So, one thing that's important here is that we _don't_ install gitops, or flux and tilt will start conciling all over each other (it's only a reconciliation the second time, right?).

You need to bootstrap to be able to follow the guide. You don't need to bootstrap to just `flux create` a few things.

Reply:
```
flux create source git podinfo \
  --url=https://github.com/stefanprodan/podinfo \
  --branch=master \
  --interval=30
```

```
flux create kustomization podinfo \
  --target-namespace=flux-system \
  --source=podinfo \
  --path="./kustomize" \
  --prune=true \
  --interval=5m
```

would suffice as an example

Reply:
agree @ozamosi but specifically in the guide we're telling users to clone the bootstrapped repo and export/commit the created flux resources

Reply:
@sympatheticmoose I added in a quick use these commands to see objects section.